docker-base/Dockerfile

```diff
@@ -17,21 +17,35 @@ RUN addgroup atlantis && \
     chmod g=u /home/atlantis/ && \
     chmod g=u /etc/passwd
 
-# Install dumb-init, gosu and git-lfs.
-ENV DUMB_INIT_VERSION=1.2.5
+# Install gosu and git-lfs.
 ENV GOSU_VERSION=1.14
 ENV GIT_LFS_VERSION=3.1.2
-RUN apk add --no-cache ca-certificates gnupg curl git unzip bash openssh libcap openssl && \
-    curl -L -s --output /bin/dumb-init "https://github.com/Yelp/dumb-init/releases/download/v${DUMB_INIT_VERSION}/dumb-init_${DUMB_INIT_VERSION}_x86_64" && \
-    chmod +x /bin/dumb-init && \
+
+# Automatically populated with the architecture the image is being built for.
+ARG TARGETPLATFORM
+
+# Install packages needed for running Atlantis.
+RUN apk add --no-cache ca-certificates curl git unzip bash openssh libcap dumb-init && \
+    # Install packages needed for building dependencies.
+    apk add --no-cache --virtual .build-deps gnupg openssl && \
     mkdir -p /tmp/build && \
     cd /tmp/build && \
-    curl -L -s --output git-lfs.tar.gz "https://github.com/git-lfs/git-lfs/releases/download/v${GIT_LFS_VERSION}/git-lfs-linux-amd64-v${GIT_LFS_VERSION}.tar.gz" && \
+
+    # git-lfs
+    curl -L -s --output git-lfs.tar.gz "https://github.com/git-lfs/git-lfs/releases/download/v${GIT_LFS_VERSION}/git-lfs-linux-${TARGETPLATFORM##*/}-v${GIT_LFS_VERSION}.tar.gz" && \
     tar -xf git-lfs.tar.gz && \
     chmod +x git-lfs && \
     mv git-lfs /usr/bin/git-lfs && \
-    curl -L -s --output gosu "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-amd64" && \
-    curl -L -s --output gosu.asc "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-amd64.asc" && \
+    git-lfs --version && \
+
+    # gosu
+    case ${TARGETPLATFORM} in \
+        "linux/amd64") GOSU_ARCH=amd64 ;; \
+        "linux/arm64") GOSU_ARCH=arm64 ;; \
+        "linux/arm/v7") GOSU_ARCH=armhf ;; \
```

Comment:
Would this `armhf` work for `linux/arm/v7`

Looks like the image does not get built.
https://github.com/tianon/gosu/blob/4c33a2fd492a38f1a3ff0bb773cd7ef2c32bef6b/Dockerfile#L42-L45

Reply:
I don't have a lot of experience with the `armhf`/`armel` variants and what they equate to, but it does seem like the `armhf` binary is built here:
https://github.com/tianon/gosu/blob/23e63902e1ade697a167e7874a5a12243ac0bae8/Dockerfile#L38-L40

I'm guessing it's compatible with an ARMv7 architecture, but I don't have a way of testing that myself, so this is a bit of an opportunistic guess, although I hadn't seen how the files were built before this, and I just saw there was binaries available on the releases page.

Reply:
yeah, I think we might need to try it ourselves.

As you can see in [here](https://zchee.github.io/golang-wiki/GoArm/),`GOARM=7` is for `armv7`

Reply:
In any case, I doubt it's any more wrong than using the binary for amd64 :)

Reply:
haha, that is true.